composer/callbacks/system_metrics_monitor.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""System metrics monitor callback."""
+
+from __future__ import annotations
+
+import logging
+import os
+
+import psutil
+import pynvml
```

Comment:
We need to import gate this for situations where the user doesn't have pynvml installed because this module will get imported whether the user uses this callback or not
```suggestion
```